src/Workspaces/Remote/ServiceHub/Services/SourceGeneration/RemoteSourceGenerationService.cs

```diff
@@ -30,12 +30,17 @@ protected override IRemoteSourceGenerationService CreateService(in ServiceConstr
     }
 
     public ValueTask<ImmutableArray<SourceGeneratedDocumentInfo>> GetSourceGeneratedDocumentInfoAsync(
-        Checksum solutionChecksum, ProjectId projectId, bool withFrozenSourceGeneratedDocuments, CancellationToken cancellationToken)
+        Checksum solutionChecksum, ProjectId projectId, bool withFrozenSourceGeneratedDocuments, bool forceOnlyRequiredDocuments, CancellationToken cancellationToken)
     {
         return RunServiceAsync(solutionChecksum, async solution =>
         {
             var project = solution.GetRequiredProject(projectId);
-            var documentStates = await solution.CompilationState.GetSourceGeneratedDocumentStatesAsync(
+
+            var compilationState = forceOnlyRequiredDocuments
+                                 ? solution.CompilationState.WithForceOnlyRequiredGenerators(projectId)
```

Comment:
Should we just be passing along the creation policy to GetSourceGeneratedDocumentStatesAsync() here rather than trying to sneak it in via another method? It feels somewhat odd to think that the policy isn't a state but something passed in, but it seems like we're kinda doing that already?

Reply:
It'd also remove forking if we have a way to say run some and then run all.

Reply:
also, please fix the indentation here :)